bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/handler/RoombaHandler.java

```diff
@@ -0,0 +1,520 @@
+                        return;
+                    }
+
+                    if (ident.ver < IdentData.MIN_SUPPORTED_VERSION) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Unsupported version " + ident.ver);
+                        return;
+                    }
+
+                    if (!ident.product.equals(IdentData.PRODUCT_ROOMBA)) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Not a Roomba: " + ident.product);
+                        return;
+                    }
+
+                    blid = ident.blid;
+                    this.blid = blid;
+                }
+
+                logger.debug("BLID is: {}", blid);
+
+                if (!config.havePassword()) {
+                    RawMQTT mqtt;
+
+                    try {
+                        mqtt = new RawMQTT(host, 8883);
+                    } catch (KeyManagementException | NoSuchAlgorithmException e1) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e1.toString());
+                        return; // This is internal system error, no retry
+                    }
+
+                    mqtt.requestPassword();
+                    RawMQTT.Packet response = mqtt.readPacket();
+                    mqtt.close();
+
+                    if (response != null && response.isValidPasswdPacket()) {
+                        RawMQTT.PasswdPacket passwdPacket = new RawMQTT.PasswdPacket(response);
+
+                        config.password = passwdPacket.getPassword();
+                        if (config.havePassword()) {
+                            Configuration configuration = editConfiguration();
+
+                            configuration.put(RoombaConfiguration.FIELD_PASSWORD, config.password);
+                            updateConfiguration(configuration);
+                        }
+                    }
+                }
+
+                if (!config.havePassword()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "Authentication on the robot is required");
+                    scheduleReconnect();
+                    return;
+                }
+
+                logger.debug("Password is: {}", config.password);
+
+                // BLID is used as both client ID and username. The name of BLID also came from Roomba980-python
+                connection = new RoombaMqttBrokerConnection(config.ipaddress, blid, this);
+                connection.start(blid, config.password);
+
+            } catch (IOException e) {
+                error = e.toString();
+            }
+
+            if (error != null) {
+                logger.error(error);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, error);
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

Reply:
I'll still leave it under debug level, it has just helped me to find and debug one issue, and it could be used to resolve possible problems on user's side.
There seems to be a firmware quirk or bug in Roomba

Reply:
This is a redundant log message. The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Reply:
Ok, removed, have fun.
By the way, what level do they use? I don't know because i never see them. At least on developer's console. But i muted a lot of stuff that deemed unrelated to me.

Reply:
Adding `<logger name="smarthome.event.ThingStatusInfoChangedEvent" level="INFO" />` to `logback.xml` should make these visible.